File: Frontend/src/interfaces/v12/colour/colour.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface ColourDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
}

export interface ColourCreateUpdateDto {
  name: string;
}

export interface ColourSearchDto extends PagedRequestAbstractDto {
  search: string;
}

File: Wms.Api/Dto/Colour/Colour.cs
 using Wms.Api.Entities; 

namespace Wms.Api.Dto.Colour
{
    public class ColourDetailsDto : CreatedChangedEntity
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
    }

    public class ColourCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class ColourSearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
}
